net.sourceforge.Teo.json

```diff
@@ -0,0 +1,75 @@
+{
+    "app-id" : "net.sourceforge.Teo",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.28",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "teo",
+    "finish-args" : [
+        /* X11 + XShm access */
+        "--share=ipc", "--socket=x11",
+        /* Play sounds */
+        "--socket=pulseaudio"
+    ],
+    "build-options" : {
+        "cflags" : "-O2 -g",
+        "cxxflags" : "-O2 -g",
+        "env" : { "DEBUGMODE": "1" }
+    },
+    "cleanup" : [
+        "/bin/*hg*",
+        "/lib/",
+        "/lib/debug/lib/",
+        "/lib/debug/source/mercurial/"
+    ],
+    "modules" : [
+        /* Remove when Mercurial support exists: Issues #834 */
+        {
+            "name" : "mercurial",
+            "buildsystem" : "simple",
+            "build-commands" : [
+                "make build",
+                "python3 setup.py install -O1 --prefix /app"
+            ],
+            "sources" : [
+                {
+                    "type" : "archive",
+                    "url" : "http://www.selenic.com/mercurial/release/mercurial-4.4.2.tar.gz",
+                    "sha256" : "dc2f72caccd6b760226753d48c2f4e8889fe176a6b23ef50775caac55ce28b85"
+                }
+            ]
+        },
+        {
+            "name" : "git-hg",
+            "buildsystem" : "simple",
+            "build-commands" : [
+                "install -m0755 git-remote-hg /app/bin/git-remote-hg"
+            ],
+            "sources" : [
+                {
+                    "type" : "file",
+                    "url" : "https://raw.githubusercontent.com/mnauw/git-remote-hg/5e96683f67af35475175bb057271102edb4a3c9a/git-remote-hg",
+                    "sha256" : "fd52d45d1ba93f6b29c7d9b2928e8c4771a21e7e06886850dc617a7239b93c43"
+                }
+            ]
+        },
+        {
+            "name" : "teo",
+            "buildsystem" : "simple",
+            "build-commands" : [
+                "./autogen.sh",
+                "make",
+                "install -m0755 teo /app/bin/teo",
+                "install -m0644 -D misc/pack/repo/linux/debian/teo/usr/share/pixmaps/net.sourceforge.Teo.png /app/share/pixmaps/net.sourceforge.Teo.png",
+                "install -m0644 -D misc/pack/repo/linux/debian/teo/usr/share/applications/net.sourceforge.Teo.desktop /app/share/applications/net.sourceforge.Teo.desktop",
+                "sed -i 's,/usr/games/,/app/bin/,' /app/share/applications/net.sourceforge.Teo.desktop",
+                "install -m0644 -D misc/pack/repo/linux/debian/teo/usr/share/metainfo/net.sourceforge.Teo.appdata.xml /app/share/metainfo/net.sourceforge.Teo.appdata.xml"
+            ],
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "hg::http://hg.code.sf.net/p/teoemulator/code"
```

Comment:
this needs to be a reproducible source. so a git tag+commit or a release archive to be available on flathub.